Conversion of Cloud API Demo bash Scripts to Python

vasujain275 opened the pull request:
**What kind of change does this PR introduce?**

This pr refactor the scripts present in `scripts/cloud-api-demo` from bash to python, this is related to https://github.com/PalisadoesFoundation/talawa-api/issues/1428 but do not completely fixes the issue.

**Issue Number:**

Fixes #1428

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

N/A

**If relevant, did you update the documentation?**

No

**Summary**

This pr converts all the important scripts needed to function talawa api's cloud instance to python. It follows all the guidelines given here - https://github.com/PalisadoesFoundation/talawa-api/issues/1428#issuecomment-1890588216

**Does this PR introduce a breaking change?**

No

**Other information**

To test these scripts on demo godaddy vps, log in as `talawa-api` user and use the following commands:

```bash
python3 deploy.py --path ~/develop/talawa-api --branch develop
```

```bash
python3 renew_certificates.py --config-dir ~/.certbot/config --logs-dir ~/.certbot/logs --work-dir ~/.certbot/work
```

```bash
python3 reset_database.py --mongo-container talawa-api_mongodb_1 --mongo-db talawa-api --repo-dir /home/talawa-api/develop/talawa-api
```

```bash
python3 correct_permissions.py --user talawa-api --files /usr/local/bin/deploy.py /usr/local/bin/reset_database.py /usr/local/bin/renew_certificates.py
```

**I have tested all the scripts on the demo vps and they are working perfectly.**


**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

palisadoes reviewed (changes requested):
1. Make the code match the Google style guide
    1. https://google.github.io/styleguide/pyguide.html#383-functions-and-methods
    ![image](https://github.com/PalisadoesFoundation/talawa-api/assets/16875803/a1b6caa6-0f53-4704-bebd-83e28ad24e34)
1. Make sure the code is pylint, python black, flake8, and pydocstyle compliant. These are pip lint and formatting packages that you can use in your IDE

palisadoes commented on scripts/cloud-api-demo/correct_permissions.py:
```diff
@@ -0,0 +1,29 @@
+    process = subprocess.run(cmd, capture_output=True, text=True)
+
+    if process.returncode != 0:
+        print(f"Error occurred: {process.stderr}")
```
If there is an error it must immediately exit with a non zero exit code after printing the message

vasujain275 replied:
Implemented

palisadoes commented on scripts/cloud-api-demo/deploy.py:
```diff
@@ -0,0 +1,42 @@
+    output, error = process.communicate()
+
+    if process.returncode != 0:
+        print(f"Error occurred: {error.decode().strip()}")
```
If there is an error it must immediately exit with a non zero exit code after printing the message

vasujain275 replied:
Implemented

palisadoes commented on scripts/cloud-api-demo/renew_certificates.py:
```diff
@@ -0,0 +1,27 @@
+    output, error = process.communicate()
+
+    if process.returncode != 0:
+        print(f"Error occurred: {error.decode().strip()}")
```
If there is an error it must immediately exit with a non zero exit code after printing the message

vasujain275 replied:
Implemented

vasujain275 commented:
@palisadoes 

Thank you for the review, I have added the changes requested that are - 
1. Linted the python files properly.
2. Added `raise SystemExit(1)` to exit with a non zero exit code after printing the error message.

palisadoes commented on scripts/cloud-api-demo/correct_permissions.py:
```diff
@@ -0,0 +1,49 @@
+import argparse
```
This must be the first line to specify the specific interpreter to use

```
#!/usr/bin/env python3
```

vasujain275 replied:
Added

palisadoes commented on scripts/cloud-api-demo/renew_certificates.py:
```diff
@@ -0,0 +1,56 @@
+import argparse
```
This must be the first line to specify the specific interpreter to use

```
#!/usr/bin/env python3
```

vasujain275 replied:
Added

palisadoes commented on scripts/cloud-api-demo/reset_database.py:
```diff
@@ -0,0 +1,63 @@
+import argparse
```
This must be the first line to specify the specific interpreter to use

```
#!/usr/bin/env python3
```

vasujain275 replied:
Added

palisadoes reviewed (changes requested):
See comments
